frontend/ui/elements/menu_keyboard_layout.lua

```diff
@@ -57,16 +57,23 @@ end
 local function genLayoutSpecificSubmenu()
     local item_table = {}
 
-    for lang, keyboard_layout in FFIUtil.orderedPairs(VirtualKeyboard.lang_to_keyboard_layout) do
-        if isKeyboardLayoutActive(lang) and VirtualKeyboard.lang_has_submenu[lang] then
+    for lang, __ in FFIUtil.orderedPairs(VirtualKeyboard.lang_has_submenu) do
+        if isKeyboardLayoutActive(lang) then
+            local keyboard_layout = VirtualKeyboard.lang_to_keyboard_layout[lang]
             local kb_pkg = "ui/data/keyboardlayouts/" .. keyboard_layout
-            local keyboard = require(kb_pkg)
-            if dbg.dassert(keyboard.genMenuItems ~= nil) then
-                table.insert(item_table, {
-                    text = Language:getLanguageName(lang),
-                    sub_item_table = keyboard:genMenuItems(),
-                })
-            end
+            table.insert(item_table, {
+                text = Language:getLanguageName(lang),
+                sub_item_table_func = function()
+                    local keyboard = require(kb_pkg)
+                    if dbg.dassert(keyboard.genMenuItems ~= nil) then
+                        return keyboard:genMenuItems()
+                    else
+                        return {
+                            text = _("Not implemented"),
+                        }
```

Comment:
Not sure what `dbg.dassert` does (crash ? warn ?), I've never seen it (and looking at the code, I'm still not sure what it does :)
It was in the original code, but do we really need to keep this curious beast ? Even more than now, you display something when it would be missing?

Reply:
OK, it's gone - didn't see your latest commit :)

Reply:
Had the exact same though, got rid of it right after that ;).

(Especially since the failure case is now handled gracefully).

(It's basically an `assert` that only runs as an `assert` in debug mode (i.e., will crash if it evaluates to false), but runs as a simple conditional if debug is not enabled).

Reply:
Hold on, what do you have against it? I'm not sure what this particular one is about bug it's quite intentional to crash in debug mode when something happens that shouldn't be happening.

Reply:
It's essentially used as "Not Yet Implemented" guard here (which... is unusual in our codebase to begin with, but, given that keyboards may be written by new contributors, okay, why not) for an *optional* feature of keyboard layout (namely, handling layout-specific options).

The original code was setting up the menu after a dassert guard, I removed it because the NYI is now handled (by adding an explicit "NYI" menu entry ;p).

Reply:
I can go back to a `dassert` and no NYI menu entry if you prefer, though?

Reply:
(I kinda not like these debug/!debug behaviour changes.
Imagine someone following the new instructions (More tools > verbose debug logs) and crashing before he gets a chance to go at reproducing his unrelated issue :)

Reply:
That's also a fair point ;).

Reply:
(Then again, I'm not a fan of *asserts*, period ;o).

Reply:
I have no special love for 'em, I just figure someone put it in for a reason. ;-)

Reply:
Ping @cyphar (since in was introduced in #8372) ;).

Reply:
The reason for making it a `dassert` is that it is a programmer error to add a language to `lang_has_submenu` without giving it a `genMenuItems` method. If these kinds of sanity checks (intended to be hit by people when writing new keyboards) are unusual in koreader, feel free to remove it.

Reply:
It's very specifically *assert*/*dassert* that I have personally not found useful, not the general principle.

Reply:
Sure.